modules/position/src/main/native/android/dalvik/AbstractMathTransform.java

```diff
@@ -0,0 +1,222 @@
+/*
+ *    Derived from:
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2001-2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    This package contains documentation from OpenGIS specifications.
+ *    OpenGIS consortium's work is fully acknowledged here.
+ */
+
+package com.gluonhq.helloandroid;
+
+import java.io.Serializable;
```

Comment:
These 3 classes for the altitude calculation don't require Android API. I was wondering if we should have them in the GraalVM/Java side, and invoke them from `AndroidPositionService` directly to modify the position readings.

Reply:
sounds like a good follow-up issue. It would be better indeed probably. I didn't really look into those classes, but if those are simply to improve the accuracy of the altitude, +1.